redbot/cogs/filter/filter.py

```diff
@@ -363,6 +363,7 @@ def __unload(self):
 
         if hits:
             try:
+                self.bot.dispatch("filter_message_delete", message)
```

Comment:
The event should only be dispatched if the deletion succeeds. To do this, move this line down to 371, just after the `else:` line :smiley:

Reply:
oh yeah, sure^^